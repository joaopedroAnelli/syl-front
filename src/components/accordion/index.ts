import { Root } from './root';
import { Header } from './header';
import { Title } from './title';
import { Indicator } from './indicator';
import { Body } from './body';
import { Description } from './description';

const Accordion = {
  Root,
  Header,
  Title,
  Indicator,
  Body,
  Description,
};

export default Accordion;
