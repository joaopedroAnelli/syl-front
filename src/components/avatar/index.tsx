import { FC } from 'react';
import Image, { ImageProps } from 'next/image';

export const Avatar: FC<ImageProps> = ({
  alt,
  width,
  height,
  className,
  ...props
}) => {
  return (
    <Image
      {...props}
      alt={alt}
      width={width || 80}
      height={height || 80}
      className={`rounded-full ${className}`}
    />
  );
};
