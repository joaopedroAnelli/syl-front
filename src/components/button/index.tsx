import React, { ButtonHTMLAttributes } from 'react';
import { buttonVariant, ButtonVariantProps } from '@/styles/button';

export type ButtonProps = ButtonHTMLAttributes<HTMLButtonElement> &
  ButtonVariantProps;

export const Button: React.FC<ButtonProps> = ({
  children,
  buttonType = 'default',
  fluid,
  ...props
}) => {
  return (
    <button
      {...props}
      className={buttonVariant({
        class: props.className,
        buttonType,
        fluid,
      })}
    >
      {children}
    </button>
  );
};
