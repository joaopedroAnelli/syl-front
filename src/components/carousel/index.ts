import { Root } from './root';
import { CarouselItem } from './item';
import { NextTrigger } from './trigger/next';
import { PrevTrigger } from './trigger/prev';

const Carousel = {
  Root,
  Item: CarouselItem,
  Trigger: {
    Next: NextTrigger,
    Prev: PrevTrigger,
  },
};

export default Carousel;
