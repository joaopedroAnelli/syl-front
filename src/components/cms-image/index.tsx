import Image, { ImageProps } from 'next/image';

import { FC } from 'react';

const CMSImage: FC<ImageProps> = ({ src, alt, ...rest }) => {
  const cmsUri = process.env.CMS_HOST || '';

  return <Image src={cmsUri + src} alt={alt} {...rest} />;
};

export default CMSImage;
