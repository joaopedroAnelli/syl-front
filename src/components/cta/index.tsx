import React, { ButtonHTMLAttributes } from 'react';
import { tv } from 'tailwind-variants';

export type CTAButtonProps = ButtonHTMLAttributes<HTMLButtonElement> & {
  buttonType?: 'primary' | 'secondary';
  fluid?: boolean;
};

const buttonVariant = tv({
  base: 'px-4 py-2 font-bold rounded-md text-lg',
  variants: {
    buttonType: {
      default: 'bg-off-white text-charcoal-grey-500',
      primary: 'bg-deep-blue text-off-white',
      secondary: 'bg-gold text-off-white',
    },
    fluid: {
      true: 'w-full',
    },
  },
  defaultVariants: {
    buttonType: 'default',
    fluid: false,
  },
});

export const CTAButton: React.FC<CTAButtonProps> = ({
  children,
  buttonType,
  fluid,
  ...props
}) => {
  return (
    <button
      {...props}
      className={buttonVariant({
        class: props.className,
        buttonType,
        fluid,
      })}
    >
      {children}
    </button>
  );
};
