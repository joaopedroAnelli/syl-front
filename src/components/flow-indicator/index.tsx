'use client';

import { FC } from 'react';
import { usePathname } from 'next/navigation';
import { FlowPage } from '@/types/flow';

export type FlowIndicatorProps = {
  flow: FlowPage[];
};

export const FlowIndicator: FC<FlowIndicatorProps> = ({ flow }) => {
  const pathname = usePathname();

  const index = flow.findIndex((f) => f.route.includes(pathname));

  return (
    <div className='w-full flex justify-center items-center gap-6'>
      {flow.map((page, i) => (
        <div
          key={page.route}
          className={`w-2 h-2 rounded-full ${
            i === index ? 'bg-deep-blue w-3 h-3' : 'bg-gray-200'
          } ${i === flow.length - 1 && i === index ? 'bg-emerald-700' : ''}`}
        />
      ))}
    </div>
  );
};
