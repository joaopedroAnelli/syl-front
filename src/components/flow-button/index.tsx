'use client';

import { FC, PropsWithChildren, useEffect } from 'react';
import { FlowPage } from '@/types/flow';
import { usePathname, useRouter } from 'next/navigation';
import { Button, ButtonProps } from '@/components/button';

type FlowButtonBaseProps = {
  flow: FlowPage[];
  isBack?: boolean;
  onClick?: () => void;
  onValidate?: () => Promise<boolean>;
};

type LinkButtonPropsWithoutHref = ButtonProps;

type FlowButtonProps = PropsWithChildren<
  FlowButtonBaseProps & LinkButtonPropsWithoutHref
>;

export const FlowButton: FC<FlowButtonProps> = ({
  flow,
  isBack = false,
  children,
  onValidate,
  ...props
}) => {
  const pathname = usePathname();

  const { push } = useRouter();

  const index = flow.findIndex((f) => f.route.includes(pathname));

  const next = isBack ? flow[index - 1] : flow[index + 1];

  const handleClick = async () => {
    let isValid = true;

    if (onValidate) {
      isValid = await onValidate();
    }

    if (!isValid) {
      return;
    }

    push(next ? next.route : '/');
  };

  return (
    <Button {...props} onClick={handleClick}>
      {children}
    </Button>
  );
};
