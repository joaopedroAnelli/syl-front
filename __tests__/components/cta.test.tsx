import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { CTAButton } from '@/components/cta';

describe('CTAButton', () => {
  it('renders a button', () => {
    render(<CTAButton href='/'>Test</CTAButton>);

    const button = screen.getByRole('link');

    expect(button).toBeInTheDocument();
  });

  it('should render with children content', () => {
    render(<CTAButton href='/'>Test</CTAButton>);

    const button = screen.getByRole('link');

    expect(button).toHaveTextContent('Test');
  });

  it('should render with properly style', () => {
    render(<CTAButton href='/'>Test</CTAButton>);

    const button = screen.getByRole('link');

    expect(button).toHaveClass(
      'px-6 py-3 font-bold rounded-md text-lg bg-off-white text-charcoal-grey-500'
    );

    expect(button).not.toHaveClass('bg-deep-blue text-off-white');
  });

  it('should render with properly style when buttonType is primary', () => {
    render(
      <CTAButton href='/' buttonType='primary'>
        Test
      </CTAButton>
    );

    const button = screen.getByRole('link');

    expect(button).toHaveClass(
      'px-6 py-3 font-bold rounded-md text-lg bg-deep-blue text-off-white'
    );
  });

  it('should render with properly style when buttonType is secondary', () => {
    render(
      <CTAButton href='/' buttonType='secondary'>
        Test
      </CTAButton>
    );

    const button = screen.getByRole('link');

    expect(button).toHaveClass(
      'px-6 py-3 font-bold rounded-md text-lg bg-gold text-deep-blue'
    );
  });

  it('should render with custom class', () => {
    render(
      <CTAButton href='/' className='custom-class'>
        Test
      </CTAButton>
    );

    const button = screen.getByRole('link');

    expect(button).toHaveClass('custom-class');
  });

  it('should render with custom class when buttonType is primary', () => {
    render(
      <CTAButton href='/' className='custom-class' buttonType='primary'>
        Test
      </CTAButton>
    );

    const button = screen.getByRole('link');

    expect(button).toHaveClass('bg-deep-blue text-off-white custom-class');
  });

  it('fluid button should take full width', () => {
    render(
      <CTAButton href='/' fluid>
        Test
      </CTAButton>
    );

    const button = screen.getByRole('link');

    expect(button).toHaveClass('w-full');
  });

  it('should run onClick function', () => {
    const onClick = jest.fn();

    render(
      <CTAButton href='/' onClick={onClick}>
        Test
      </CTAButton>
    );

    const button = screen.getByRole('link');

    button.click();

    expect(onClick).toHaveBeenCalled();
  });

  it('should match snapshot', () => {
    const { container } = render(<CTAButton href='/'>Test</CTAButton>);

    expect(container).toMatchSnapshot();
  });
});
