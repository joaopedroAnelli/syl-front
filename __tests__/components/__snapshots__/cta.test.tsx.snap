// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CTAButton should match snapshot 1`] = `
<div>
  <a
    class="px-6 py-3 font-bold rounded-md text-lg bg-off-white text-charcoal-grey-500 block text-center"
    href="/"
  >
    Test
  </a>
</div>
`;
